worker/src/RTC/KeyFrameRequestManager.cpp

```diff
@@ -101,6 +101,9 @@ void RTC::KeyFrameRequestManager::OnKeyFrameRequestTimeout(PendingKeyFrameInfo*
 
 	MS_ASSERT(it != this->mapSsrcPendingKeyFrameInfo.end(), "PendingKeyFrameInfo not present in the map")
 
+	// Best effort in case the PLI/FIR was lost
+	this->listener->OnKeyFrameNeeded(pendingKeyFrameInfo->GetSsrc());
```

Comment:
NOTE: Better read the ssrc before into a variable and use it. And then invoke the listener at the end of the function. Why? Just imagine that the listener (Producer) decides to delete the `KeyFrameManager` within the `OnKeyFrameNeeded` cb. It would call the destructor which would delete all the `pendingKeyFrameInfos` (including this one) and the next line would crash.

It's safe to place all the calls to listener's callback at the end of the methods.